server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/CodecFactory.java

```diff
@@ -80,6 +81,24 @@ public CompressionCodec getCodec(String name, Configuration conf) {
         return codecClass;
     }
 
+    /**
+     * Determine whether a given compression codec is safe for multiple concurrent threads
+     *
+     * @param compCodec     the user-given COMPRESSION_CODEC, may be null
+     * @param dataSource    the file that we are accessing
+     * @param configuration HDFS config
+     * @return true only if it's thread safe
+     */
+    public boolean isCodecThreadSafe(String compCodec, String dataSource, Configuration configuration) {
+        if (compCodec == null) {
+            // check for file extensions indicating bzip2 (Text only)
+            // currently doesn't check for bzip2 in .avro files
+            Class<? extends CompressionCodec> codecClass = getCodecClassByPath(configuration, dataSource);
+            return (codecClass == null || !BZip2Codec.class.isAssignableFrom(codecClass));
+        }
+        return !compCodec.equalsIgnoreCase("bzip2") && !compCodec.equalsIgnoreCase(BZip2Codec.class.getName());
```

Comment:
I was suggesting to collapse into 1 return statement, like this:

```suggestion
        Class<? extends CompressionCodec> codecClass = null;
        if (compCodec == null) {
            // check for file extensions indicating bzip2 (Text only)
            // currently doesn't check for bzip2 in .avro files
            codecClass = getCodecClassByPath(configuration, dataSource);
        }
        return !( "bzip2".equalsIgnoreCase(compCodec) ||
                       BZip2Codec.class.getName().equalsIgnoreCase(compCodec) ||
                       (codecClass != null && BZip2Codec.class.isAssignableFrom(codecClass))
                      );
```

Reply:
This looks equivalent, I will give it a try.